gemrb/core/GameData.cpp

```diff
@@ -268,23 +235,22 @@ Effect* GameData::GetEffect(const ResRef &resname)
 		return nullptr;
 	}
 
-	effect = em->GetEffect();
-	if (effect == nullptr) {
+	auto newEffect = em->GetEffect();
+	if (newEffect == nullptr) {
 		return nullptr;
 	}
 
-	EffectCache.SetAt(resname, (void *) effect);
-	return new Effect(*effect);
+	EffectCache.SetAt(resname, std::move(*newEffect));
+
+	auto effectCopy = new Effect(*newEffect);
```

Comment:
Wouldn't it be safer/more future-proof to create the copy before the std::move?

Reply:
Good catch